Content.Client/Flash/FlashOverlay.cs

```diff
@@ -49,21 +50,27 @@ public void ReceiveFlash(double duration)
             _lastsFor = duration;
         }
 
-        protected override void Draw(in OverlayDrawArgs args)
+        protected override bool BeforeDraw(in OverlayDrawArgs args)
         {
             if (!_entityManager.TryGetComponent(_playerManager.LocalEntity, out EyeComponent? eyeComp))
-                return;
-
+                return false;
             if (args.Viewport.Eye != eyeComp.Eye)
-                return;
+                return false;
 
-            var percentComplete = (float) ((_gameTiming.CurTime.TotalSeconds - _startTime) / _lastsFor);
-            if (percentComplete >= 1.0f)
-                return;
+            _percentComplete = (float) ((_gameTiming.CurTime.TotalSeconds - _startTime) / _lastsFor);
+            if (_percentComplete >= 1.0f)
+            {
+                _screenshotTexture = null;
+                return false;
+            }
+            return true;
+        }
 
+        protected override void Draw(in OverlayDrawArgs args)
+        {
             var worldHandle = args.WorldHandle;
+            _shader.SetParameter("percentComplete", _percentComplete);
             worldHandle.UseShader(_shader);
```

Comment:
This ordering doesn't matter until the draw call is done.

Reply:
The order does not matter, but I was just making it consistent with how other overlays do it. See for example [DrunkOverlay.cs](https://github.com/space-wizards/space-station-14/blob/master/Content.Client/Drunk/DrunkOverlay.cs) or [RainbowOverlay.cs](https://github.com/space-wizards/space-station-14/blob/master/Content.Client/Drugs/RainbowOverlay.cs).